### DISCUSSION TITLE
Docker agent "no-docker-interface" no longer available after update to 0.15.0?

### DISCUSSION THREAD

**Discussion by mr-tester-william-lo** [PrefectHQ/prefect#4757] [Q&A] [answered]
Previous version 0.14.22 (work well when run the docker flow)

```
FROM prefecthq/prefect:0.14.22
RUN prefect backend server
CMD prefect agent docker start -a $API_URL \
--volume /tmp/test_folder:/opt/test_folder:ro \
--name I_AM_GOOD_DOCKER_AGENT --no-docker-interface
```

After update version 0.15.0 (error message, failed to run docker flow )
```
FROM prefecthq/prefect:0.15.0
RUN prefect backend server
CMD prefect agent docker start -a $API_URL \
--volume /tmp/test_folder:/opt/test_folder:ro \
--name I_AM_GOOD_DOCKER_AGENT --no-docker-interface
```

Error Message:
400 Client Error for http+docker://localhost/v1.40/containers/create?name=stylish-dolphin: Bad Request ("invalid IP address in add-host: "host-gateway"")

What should I do?

**Comment by zanieb** [comment, marked as answer]
`--no-docker-interface` was deprecated in https://github.com/PrefectHQ/prefect/pull/4446 and has had no affect since 0.14.18

https://github.com/PrefectHQ/prefect/pull/4714 added the "host-gateway" fix for some Linux users, which appears to be causing you an error. Can you share the output of `docker version`?

**Reply by mr-tester-william-lo** [reply]
```
Client:
 Version:           19.03.6-ce
 API version:       1.40
 Go version:        go1.13.4
 Git commit:        369ce74
 Built:             Fri May 29 04:01:26 2020
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.6-ce
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.4
  Git commit:       369ce74
  Built:            Fri May 29 04:01:57 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.3.2
  GitCommit:        ff48f57fc83a8c44cf4ad5d672424a98ba37ded6
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

**Reply by zanieb** [reply]
I believe this would be resolved by upgrading to Docker v20.10.0 or newer as those versions of Docker gained the "host-gateway" inference (https://github.com/moby/moby/pull/40007/commits/92e809a6807210a3d1ecd7949314367e82f5b683). If enough people encounter this, we could add an explicit check of the docker version before adding the host but that will just introduce more issues across versions. I think the best path forward is to upgrade Docker.

**Reply by mr-tester-william-lo** [reply]
thanks so much 
problem solved by upgrade the Docker from 19 to 20.10
it will be great if there are suggested versioning in doc in future

**Reply by zanieb** [reply]
Yep! https://github.com/PrefectHQ/prefect/pull/4772 adds this :)
